woocommerce/payment-gateway/payment-tokens/class-sv-wc-payment-gateway-payment-tokens-handler.php

```diff
@@ -584,6 +584,35 @@ public function update_tokens( $user_id, $tokens, $environment_id = null ) {
 	}
 
 
+	/**
+	 * Updates a single legacy token in user meta.
+	 *
+	 * @internal
+	 *
+	 * @since 5.6.0-dev.1
+	 *
+	 * @param int $user_id WP user ID
+	 * @param SV_WC_Payment_Gateway_Payment_Token $token token to update
+	 * @param string|null $environment_id optional environment ID, defaults to plugin current environment
+	 * @return string|int updated user meta ID
+	 */
+	public function update_legacy_token( $user_id, $token, $environment_id = null ) {
+
+		// default to current environment
+		if ( null === $environment_id ) {
+			$environment_id = $this->get_environment_id();
+		}
+
+		$legacy_tokens = get_user_meta( $user_id, $this->get_user_meta_name( $environment_id ), true );
```

Comment:
Even if `$legacy_tokens` is `false` (default return value here), I believe PHP type juggling should be able to convert to array and let us set the legacy token as you do right below. However, I generally don't trust data coming from `get_user_meta()` since it can be of any type and I'd default `$legacy_tokens`  to array type if it's not one.